### PR TITLE
fix: crontab with empty schedule

### PR DESCRIPTION
**Pull request by timmilesdw**
#### Overview

This PR improves the `scheduleManager` component by adding input validation, error handling, and thread safety.

#### What this PR does / why we need it

- Prevents adding schedule entries with empty crontab strings, logging an error instead.
- Handles and logs errors from invalid crontab expressions when adding new cron jobs.
- Adds a mutex to the `scheduleManager` to ensure thread-safe access to the internal entries map, preventing race conditions during concurrent operations.
- Updates and stabilizes tests to cover edge cases and concurrent usage.

When crontab is empty, we skip it and log error
```json
{"level":"error","msg":"crontab is empty","stacktrace": ... ,"time":"2025-07-04T19:39:28+03:00"}
```
When crontab is invalid, we skip it and log error
```json
{"level":"error","msg":"invalid crontab","crontab":"stub-crontab","stacktrace": ... ,"time":"2025-07-04T19:39:28+03:00"}
```

These changes make the scheduling logic more reliable and easier to maintain, and ensure that invalid or concurrent usage does not cause subtle bugs.

### CODE REVIEW

File: pkg/schedule_manager/schedule_manager.go [hunk 1]
```diff
@@ -59,18 +61,28 @@ func (sm *scheduleManager) Stop() {
 // Crontab string should be validated with cron.Parse
 // function before pass to Add.
 func (sm *scheduleManager) Add(newEntry smtypes.ScheduleEntry) {
+	if newEntry.Crontab == "" {
```
Comment (ldmonster) [inline pkg/schedule_manager/schedule_manager.go:64]: looks like we can move it under 69 line and just call logEntry.Error()

File: pkg/schedule_manager/schedule_manager.go [hunk 2]
```diff
@@ -59,18 +61,28 @@ func (sm *scheduleManager) Stop() {
...
+		sm.logger.With("operator.component", "scheduleManager").
+			Error("crontab is empty", slog.String("name", newEntry.Crontab))
```
Comment (ldmonster) [inline pkg/schedule_manager/schedule_manager.go:66]:
```suggestion
			Error("crontab is empty", slog.String("crontab", newEntry.Crontab))
```

File: pkg/schedule_manager/schedule_manager.go [hunk 3]
```diff
@@ -59,18 +61,28 @@ func (sm *scheduleManager) Stop() {
...
+
 	cronEntry, hasCronEntry := sm.Entries[newEntry.Crontab]
 
 	// If no entry, then add new scheduled function and save CronEntry.
 	if !hasCronEntry {
-		// The error can occur in case of bad format of crontab string.
-		// All crontab strings should be validated before add.
-		entryId, _ := sm.cron.AddFunc(newEntry.Crontab, func() {
+		entryId, err := sm.cron.AddFunc(newEntry.Crontab, func() {
 			logEntry.Debug("fire schedule event for entry", slog.String("name", newEntry.Crontab))
```
Comment (ldmonster) [inline pkg/schedule_manager/schedule_manager.go:79]:
```suggestion
			logEntry.Debug("fire schedule event for entry", slog.String("crontab", newEntry.Crontab))
```

**Review by ldmonster — APPROVED** [review]

**Review by ldmonster — APPROVED** [review]
